libsolidity/ast/Types.cpp

```diff
@@ -406,44 +578,59 @@ bool IntegerConstantType::isImplicitlyConvertibleTo(Type const& _convertTo) cons
 			return true;
 		return false;
 	}
+	else if (_convertTo.category() == Category::FixedPoint)
+	{
+		if (fixedPointType() && fixedPointType()->isImplicitlyConvertibleTo(_convertTo))
```

Comment:
We cannot use `fixedPointType()` here because we have to return false if the number does not fit into finite precision. Instead, I would propose to do something like
```
rational shifted = (m_value * (bigint(1) << (8 * _convertTo.fractionalBits())));
if (shifted.denominator() != 1) return false;
// here, do the same stuff we do above for integers with _convertTo.integerBits() + _convertTo.fractioalBits() bits.
```